src/chainstate/burn/db/sortdb.rs

```diff
@@ -619,6 +645,67 @@ impl <'a> SortitionHandleTx <'a> {
         self.set_stacks_block_accepted_at_tip(&chain_tip, consensus_hash, parent_stacks_block_hash, stacks_block_hash, stacks_block_height)
     }
 
+    fn pick_recipients(&self, reward_set_vrf_seed: &SortitionHash, next_pox_info: Option<&RewardCycleInfo>) -> Result<Option<RewardSetInfo>, BurnchainError> {
+        if let Some(next_pox_info) = next_pox_info {
+            if let PoxAnchorBlockStatus::SELECTED_AND_KNOWN(ref anchor_block, ref reward_set) = next_pox_info.anchor_status {
+                if reward_set.len() == 0 {
+                    return Ok(None)
+                }
+
+                let chosen_recipients = reward_set_vrf_seed.choose(OUTPUTS_PER_COMMIT.try_into().unwrap(),
+                                                                   reward_set.len().try_into().unwrap());
+                Ok(Some(RewardSetInfo {
+                    anchor_block: anchor_block.clone(),
+                    recipients: chosen_recipients.into_iter().map(|ix| {
+                        let recipient = reward_set[ix as usize].clone();
+                        (recipient, u16::try_from(ix).unwrap())
```

Comment:
Should we `expect`, instead of `unwrap`?

Reply:
Sure -- updated in 4a6a38a30